apps/landing-page/pages/blog/using-openshift-authentication-to-secure-access-to-backstage.mdx

```diff
@@ -0,0 +1,177 @@
+2. TLS certificates to expose the OAuth container
+3. Configure the Route that is exposed by the OpenShift ingress router
+
+Let’s describe how to enable each of these configurations.
+
+### OAuth Client Configuration
+
+OpenShift provides two methods for registering a new OAuth client:
+
+1. Creating a OAuthClient Custom Resource
+2. Using a Service Account as an OAuth client
+
+The former is commonly used by cluster services, such as the OpenShift web console, but has the disadvantage of a cluster scoped resource, thus requiring elevated access. The latter is more applicable in this use case as a standard OpenShift Service Account can perform the function of acting as an OAuth client. All that is required to enable such functionality is adding an annotation with the key `serviceaccounts.openshift.io/oauth-redirecturi.<name>` to the Service Account with the location of the redirect URI. For the case of the OpenShift OAuth proxy, it would be at the `/oauth/callback` context path.
+
+This capability can be enabled by setting the following Helm values.
+
+```yaml
+serviceAccount:
+  create: true
+  annotations:
+    serviceaccounts.openshift.io/oauth-redirecturi.backstage: "https://{{ .Values.ingress.host }}/oauth/callback"
+```
+
+### TLS Certificate Management
+
+There are multiple methods for which TLS certificates can be configured within OpenShift. They could be provided by the end user or by way of an operator, such as [cert-manager](https://docs.openshift.com/container-platform/4.12/security/cert_manager_operator/cert-manager-operator-install.html), which can integrate with an external certificate management tool or generate certificates of its own. OpenShift also includes support for automatically generating and injecting certificates into applications through the [Service Serving Certificate](https://docs.openshift.com/container-platform/4.12/security/certificates/service-serving-certificate.html) feature. The `service-ca` monitors annotations placed upon OpenShift `Service` resources. When a Service with the annotation `service.beta.openshift.io/serving-cert-secret-name=&lt;name>` is created, the controller generates a certificate and its associated private key within a `Secret`.
+
+To have OpenShift generate certificate within a Secret called `backstage-tls` which can be configured within the OAuth proxy, the following Helm values can be specified:
+
+```yaml
+service:
+  annotations:
+    service.alpha.openshift.io/serving-cert-secret-name: backstage-tls
+```
+
+### Ingress Configuration
+
+The ability to expose applications running within OpenShift externally has been one of the most compelling features included within OpenShift ever since the early days of version 3. Routes, the predecessor to the native Kubernetes Ingress resource, has enabled such functionality and it continues to this day. However, more and more applications are favoring the native Kubernetes Ingress option over the OpenShift specific Route.
```

Comment:
```suggestion
The ability to expose applications running within OpenShift externally has been one of the most compelling features included within OpenShift ever since the early days of version 3. `Routes`, the predecessor to the native Kubernetes `Ingress` resource, has enabled such functionality and it continues to this day. However, more and more applications are favoring the native Kubernetes `Ingress` option over the OpenShift specific `Route`.
```